### PR TITLE
feat: template (archived)

### PR DESCRIPTION
**Pull request by pichoemr**
#__ 

## Checklist
- [ ] I didn't over-scope my PR
- [ ] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [ ] I added/updated the documentation about my changes
- [ ] I made my own code-review before requesting one

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+$manifest = array(
+  "widgets" => array(
```
Comment (taorepoara) [inline manifest.php]: Widgets and listeners are deprecated from the manifest

File: .lenra/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# syntax=docker/dockerfile:1.4
```
Comment (taorepoara) [inline .lenra/Dockerfile]: The `.lenra` directory should be ignored

**Review by taorepoara — CHANGES_REQUESTED** [review]

File: index.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+require 'vendor/autoload.php';
+
+include "manifest.php";
+
+header("Content-Type: application/json");
+
+$app = AppFactory::create();
+
+$app->post('/function/{appUuid}', function (Request $request, Response $response, array $args) {
```
Comment (taorepoara) [inline index.php]: What does the `function/{appUuid}` is made for ?

File: index.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    global $manifest;
+    $body = json_decode($request->getBody(), true);
+
+    if (isset($body["resource"])) {
+        $image = file_get_contents('./resources/' . $body["resource"]);
+        $response->getBody()->write($image);
+        return $response->withHeader('Content-Type', 'img/png');
```
Comment (taorepoara) [inline index.php]: Is the content type needed ? If yes, you should determine it from the file extension, if not remove it

File: listeners/increment.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+require "./services/api.php";
+
+function run($props, $event, $api)
+{
+    $counter = json_decode(getDoc($api, "counter", $props["id"])->getBody(), true);
```
Comment (taorepoara) [inline listeners/increment.php]: You should decode the JSON in the `getDoc` method:

```suggestion
    $counter = getDoc($api, "counter", $props["id"]);
```

File: listeners/onEnvStart.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<?php
+require "./services/api.php";
+
+function run($props, $event, $api)
+{
+    $res = executeQuery($api, "counter", array(
+        "user" => "global"
+    ));
+
+    $counters = json_decode($res->getBody());
```
Comment (taorepoara) [inline listeners/onEnvStart.php]: You should decode the JSON in the `executeQuery` method:

```suggestion
    $counters = executeQuery($api, "counter", array(
        "user" => "global"
    ));
```

File: listeners/onUserFirstJoin.php [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?php
+require "./services/api.php";
+
+function run($props, $event, $api)
+{
+    $res = executeQuery($api, "counter", array(
+        "user" => "@me"
+    ));
+
+    $counters = json_decode($res->getBody());
```
Comment (taorepoara) [inline listeners/onUserFirstJoin.php]: Same as erlier:
```suggestion
    $counters = executeQuery($api, "counter", array(
        "user" => "@me"
    ));
```

**Review by taorepoara — CHANGES_REQUESTED** [review]

File: index.php (hunk 3 above)
Comment (pichoemr) [reply index.php]: To accept post requests on /function/{appUuid}, {appUuid} is a parameter of the url so we accept requests for any appUuid passed, we also accept /fucntion/test for dev-tool requests.

Comment (taorepoara) [reply index.php]: Why does the dev-tool call the app with this path ? Can you look into the dev-tool please ?

**Comment by taorepoara** [comment]
You should add informations to the `README.md` just like in the Rust template.

**Comment by taorepoara** [comment]
I have this error when I try to use the template: `[200]: POST /function/test - Uncaught Error: Failed opening required 'vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /app/index.php:7`

File: index.php (hunk 3 above)
Comment (taorepoara) [reply index.php]: Should be replaced with a wildcard:

```suggestion
$app->post('/{path:.*}', function (Request $request, Response $response, array $args) {
```

File: lenra.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+generator:
+  dofigen:
+    builders:
+      - name: builder
+        image: composer:latest
+        adds:
+          - "."
+        workdir: /app
+        script:
+          - composer
```
Comment (taorepoara) [inline lenra.yml]:
```suggestion
          - composer i
```

**Review by taorepoara — CHANGES_REQUESTED** [review]

**Comment by pichoemr** [comment]
@taorepoara  I test `lenra check template` on the template, i got an error:
```
manifest            : Error
    manifest:unexpectedError
        Error loading manifest checker data: Request(Transport(Transport { kind: ConnectionFailed, message: Some("Connect error"), url: Some(Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8080), path: "/", query: None, fragment: None }), source: Some(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }))
widget:main         : Error
    widget:main:unexpectedError
        Error loading widget:main checker data: Request(Transport(Transport { kind: ConnectionFailed, message: Some("Connect error"), url: Some(Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8080), path: "/", query: None, fragment: None }), source: Some(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }))
widget:menu         : Error
    widget:menu:unexpectedError
        Error loading widget:menu checker data: Request(Transport(Transport { kind: ConnectionFailed, message: Some("Connect error"), url: Some(Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8080), path: "/", query: None, fragment: None }), source: Some(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }))
widget:home         : Error
    widget:home:unexpectedError
        Error loading widget:home checker data: Request(Transport(Transport { kind: ConnectionFailed, message: Some("Connect error"), url: Some(Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8080), path: "/", query: None, fragment: None }), source: Some(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }))
widget:counter      : Error
    widget:counter:unexpectedError
        Error loading widget:counter checker data: Request(Transport(Transport { kind: ConnectionFailed, message: Some("Connect error"), url: Some(Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8080), path: "/", query: None, fragment: None }), source: Some(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }) }))
```

**Review by taorepoara — DISMISSED** [review]

File: README.md [hunk 9]
```diff
@@ -1,2 +1,108 @@
...
+
+To run the Rust template you first need to build the application.
+```console
+lenra build
+```
+
+Then you can start it.
+```console
+lenra start
+```
```
Comment (taorepoara) [inline README.md]: Maybe replace the `build` and `start` by the `dev` command ?

```suggestion
To run the PHP template you can use the `dev` command that will build and start your app and enter in a custom terminal:

```console
lenra dev
```
```

**Review by taorepoara — CHANGES_REQUESTED** [review]

**Comment by taorepoara** [comment]
Merge this and edit the README to say the project is deprecated and archived to to use the FrankenPHP template

**Pull request merged**